From ed1f943a989bb855785556222e2570c05e8f6d89 Mon Sep 17 00:00:00 2001
From: Paul Thomson <paulthomson@google.com>
Date: Fri, 17 Apr 2026 11:08:44 +0100
Subject: [PATCH 3/5] Avoid getMirror() to skip decompilation

..which is unavailable from CLI.

I cannot find any related info on why this was introduced, although the
change is fairly self-explanatory.
---
 .../uast/kotlin/internal/baseKotlinInternalUastUtils.kt       | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/plugins/kotlin/uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/internal/baseKotlinInternalUastUtils.kt b/plugins/kotlin/uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/internal/baseKotlinInternalUastUtils.kt
index 03f53460f199..11d3f79ab81a 100644
--- a/plugins/kotlin/uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/internal/baseKotlinInternalUastUtils.kt
+++ b/plugins/kotlin/uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/internal/baseKotlinInternalUastUtils.kt
@@ -80,7 +80,9 @@ inline fun <reified T : UDeclaration, reified P : PsiElement> unwrap(element: P)
 fun unwrapFakeFileForLightClass(file: PsiFile): PsiFile {
     return when (file) {
         is FakeFileForLightClass -> file.ktFile
-        is PsiCompiledElement -> (file.mirror as? KtFile) ?: file
+        is PsiCompiledElement ->
+          // .mirror triggers decompilation, which is not available in CLI
+          file // (file.mirror as? KtFile) ?: file
         else -> file
     }
 }
-- 
2.54.0.545.g6539524ca2-goog

